packages/backend-core/src/sql/utils.ts

```diff
@@ -66,6 +66,14 @@ export function buildExternalTableId(datasourceId: string, tableName: string) {
   return `${datasourceId}${DOUBLE_SEPARATOR}${tableName}`
 }
 
+export function checkTableId(tableId: string) {
```

Comment:
Should this not be encoding always? It will encode only if it is actually needed

Reply:
true - I can adjust them both - I was just replicating how it builds the table IDs today!